d2compiler/compile.go

```diff
@@ -735,8 +735,20 @@ func (c *compiler) validateNear(g *d2graph.Graph) {
 					c.errorf(obj.Attributes.NearKey, "constant near keys can only be set on root level shapes")
 					continue
 				}
-				if len(obj.ChildrenArray) > 0 {
-					c.errorf(obj.Attributes.NearKey, "constant near keys cannot be set on shapes with children")
+
+				var edgeConnectToOutsideOfNear *d2graph.Edge
+				for _, edge := range g.Edges {
+					srcNearContainer := edge.Src.OuterNearContainer()
+					dstNearContainer := edge.Dst.OuterNearContainer()
+
+					if srcNearContainer != dstNearContainer {
+						edgeConnectToOutsideOfNear = edge
+						break
```

Comment:
don't break. we want to get all the errors

Reply:
fixed